### PR TITLE
SONAR-6565 refactor users page

### PR DESCRIPTION
**Pull request by stas-vilchik**
*Note: This is a long-life PR.*

### CODE REVIEW

File: server/sonar-web/src/main/js/apps/users/header-view.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      'click #users-create': 'onCreateClick'
+    },
+
+    onCreateClick: function (e) {
+      e.preventDefault();
+      this.createUser();
+    },
+
+    createUser: function () {
+      new CreateView({
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/apps/users/header-view.js]: NewMAJORReplace by a standard call to the function.[rule](http://nemo.sonarqube.org/coding_rules#rule_key=javascript%3AConstructorFunctionsForSideEffects)

File: server/sonar-web/src/main/js/apps/users/deactivate-view.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+define([
+  'components/common/modal-form',
+  './templates'
+], function (ModalForm) {
+
+  return ModalForm.extend({
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/apps/users/deactivate-view.js]: NewBLOCKERAvoid trailing comma in array and object literals.[rule](http://nemo.sonarqube.org/coding_rules#rule_key=javascript%3ATrailingComma)

File: server/sonar-web/src/main/js/apps/users/list-item-view.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+    updateUser: function () {
+      new UpdateView({
+        model: this.model,
+        collection: this.model.collection
+      }).render();
+    },
+
+    changePassword: function () {
+      new ChangePasswordView({
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/apps/users/list-item-view.js]: NewMAJORReplace by a standard call to the function.[rule](http://nemo.sonarqube.org/coding_rules#rule_key=javascript%3AConstructorFunctionsForSideEffects)

File: server/sonar-web/src/main/js/apps/users/list-item-view.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+      this.changePassword();
+    },
+
+    onDeactivateClick: function (e) {
+      e.preventDefault();
+      this.deactivateUser();
+    },
+
+    updateUser: function () {
+      new UpdateView({
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/apps/users/list-item-view.js]: NewMAJORReplace by a standard call to the function.[rule](http://nemo.sonarqube.org/coding_rules#rule_key=javascript%3AConstructorFunctionsForSideEffects)

File: server/sonar-web/src/main/js/apps/users/list-item-view.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+    changePassword: function () {
+      new ChangePasswordView({
+        model: this.model,
+        collection: this.model.collection
+      }).render();
+    },
+
+    deactivateUser: function () {
+      new DeactivateView({ model: this.model }).render();
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/apps/users/list-item-view.js]: NewMAJORReplace by a standard call to the function.[rule](http://nemo.sonarqube.org/coding_rules#rule_key=javascript%3AConstructorFunctionsForSideEffects)

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 new issues:
* 1 blocker
* 4 major

Details:
* [Avoid trailing comma in array and object literals.](https://github.com/SonarSource/sonarqube/blob/c3a011df776427208a24b5620f51ffbcd9752fc9/server/sonar-web/src/main/js/apps/users/deactivate-view.js#L6)[rule](http://nemo.sonarqube.org/coding_rules#rule_key=javascript%3ATrailingComma)
* [Replace by a standard call to the function.](https://github.com/SonarSource/sonarqube/blob/c3a011df776427208a24b5620f51ffbcd9752fc9/server/sonar-web/src/main/js/apps/users/header-view.js#L19)[rule](http://nemo.sonarqube.org/coding_rules#rule_key=javascript%3AConstructorFunctionsForSideEffects)
* [Replace by a standard call to the function.](https://github.com/SonarSource/sonarqube/blob/c3a011df776427208a24b5620f51ffbcd9752fc9/server/sonar-web/src/main/js/apps/users/list-item-view.js#L57)[rule](http://nemo.sonarqube.org/coding_rules#rule_key=javascript%3AConstructorFunctionsForSideEffects)
* [Replace by a standard call to the function.](https://github.com/SonarSource/sonarqube/blob/c3a011df776427208a24b5620f51ffbcd9752fc9/server/sonar-web/src/main/js/apps/users/list-item-view.js#L50)[rule](http://nemo.sonarqube.org/coding_rules#rule_key=javascript%3AConstructorFunctionsForSideEffects)
* [Replace by a standard call to the function.](https://github.com/SonarSource/sonarqube/blob/c3a011df776427208a24b5620f51ffbcd9752fc9/server/sonar-web/src/main/js/apps/users/list-item-view.js#L43)[rule](http://nemo.sonarqube.org/coding_rules#rule_key=javascript%3AConstructorFunctionsForSideEffects)

**Comment by simonbrandhof** [comment]
@stas-vilchik what does "long-life PR" mean ? It should not be reviewed ?

**Comment by stas-vilchik** [comment]
That means I expect some more commits here.

**Comment by simonbrandhof** [comment]
so why did you create the PR ? The branch is enough.

**Comment by stas-vilchik** [comment]
PR is quite big, I want to benefit from the analysis and potential review ASAP.

**Comment by simonbrandhof** [comment]
Ok, so you expect someone to review. It was not clear.

File: server/sonar-web/src/main/js/apps/users/change-password-view.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+      ModalForm.prototype.onFormSubmit.apply(this, arguments);
+      this.sendRequest();
+    },
+
+    sendRequest: function () {
+      var that = this,
+          password = this.$('#change-user-password-password').val(),
+          confirmation = this.$('#change-user-password-password-confirmation').val();
+      if (password !== confirmation) {
+        that.showErrors([{ msg: 'New password and its confirmation do not match' }]);
```
Comment (jblievremont) [inline server/sonar-web/src/main/js/apps/users/change-password-view.js]: Missing I18n here.

File: server/sonar-web/src/main/js/apps/users/templates/users-change-password.hbs [hunk 7]
```diff
@@ -0,0 +1,24 @@
+<form id="change-user-password-form" autocomplete="off">
+  <div class="modal-head">
+    <h2>Change Password</h2>
```
Comment (jblievremont) [inline server/sonar-web/src/main/js/apps/users/templates/users-change-password.hbs]: Missing I18n

**Comment by stas-vilchik** [comment]
I missed l18n especially because this is an admin page.

File: server/sonar-web/src/main/js/apps/users/templates/users-deactivate.hbs [hunk 8]
```diff
@@ -0,0 +1,13 @@
+<form id="deactivate-user-form" autocomplete="off">
+  <div class="modal-head">
+    <h2>Deactivate User</h2>
```
Comment (jblievremont) [inline server/sonar-web/src/main/js/apps/users/templates/users-deactivate.hbs]: Missing I18n

File: server/sonar-web/src/main/js/apps/users/templates/users-form.hbs [hunk 9]
```diff
@@ -0,0 +1,52 @@
+<form id="create-user-form" autocomplete="off">
+  <div class="modal-head">
+    <h2>{{#if login}}Update{{else}}Create{{/if}} User</h2>
```
Comment (jblievremont) [inline server/sonar-web/src/main/js/apps/users/templates/users-form.hbs]: Missing I18n

File: server/sonar-web/src/main/js/apps/users/users.js [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+
+  return Backbone.Collection.extend({
+    model: User,
+
+    url: function () {
+      return baseUrl + '/api/users/search';
+    },
+
+    parse: function (r) {
+      this.total = +r.total;
```
Comment (jblievremont) [inline server/sonar-web/src/main/js/apps/users/users.js]: Why is there a ``+`` in this expression?

Comment (stas-vilchik) [inline server/sonar-web/src/main/js/apps/users/users.js]: to cast value to a number (I do some arithmetic with the value further)

**Comment by jblievremont** [comment]
OK, sounds good until here :+1:

**Pull request closed without merging**